src/main/scala/com/lucidchart/open/relate/SqlStatement.scala

```diff
@@ -18,33 +18,20 @@ import scala.reflect.ClassTag
 class SqlStatement(val stmt: PreparedStatement, val names: scala.collection.Map[String, List[Int]],
   val listParams: mutable.Map[String, ListParam]) {
 
-  private def list[A](name: String, values: TraversableOnce[A], rule: (Int, A) => Unit) {
-    val valueIterator = values.toIterator
-    val nameData = names.get(name)
-
-    if (nameData.isDefined) {
-      var i = 0
-      while (valueIterator.hasNext) {
-        val value = valueIterator.next
-        val parameterIterator = nameData.get.toIterator
-
-        while (parameterIterator.hasNext) {
-          rule(parameterIterator.next + i, value)
+  private def list[A](name: String, values: TraversableOnce[A], rule: (Int, A) => Unit): Unit = {
+    names.get(name).map { nameData: List[Int] =>
+      values.toSeq.zipWithIndex.map { zip: (A, Int) =>
```

Comment:
Also, rather than capturing the variable zip and then using a match statement you can just use:

```scala
values.toIterator.zipWithIndex.map { case (value, index) => 
  nameData.foreach { nameDataIndex: Int =>
    rule(nameDataIndex + index, value)
  }
}
```

Also note the use of foreach instead of map. Again, this avoids creating a new data structure that isn't used.